### DISCUSSION TITLE
How to control size of Logs of Spark SQL Engine in kyubbi

### DISCUSSION THREAD

**Discussion by kmsantanu** [apache/kyuubi#3668] [Q&A]
engine log in "$KYUUBI_WORK_DIR_ROOT/${session username}/kyuubi-spark-sql-engine.log.${num} ". Different session users have different folders to group all live and historical engine logs. Each engine will have one and only engine log.

In production,i want to keep max 100mb of log for last 3 files ,How can I doit.

currently this single file grow like in 20 to 25 gb perday.It is eating of my spaces.

I did not find anything in document.kindly suugest on it.

**Comment by pan3793** [comment]
Unfortunately, Kyuubi does not support configuring it now, you can find the detailed implementation at `org.apache.kyuubi.engine.ProcBuilder`

> keep max 100mb of log ...

I think this requirement is reasonable

> ... of log for last 3 files

Kyuubi will try its best to reuse the existing files, if you open 4 engines at the same time, there should be at least exist 4 files.

**Comment by yaooqinn** [comment]
try `logrotate`

**Comment by paprasan** [comment]
Hi i am facing a similar challenge can you point me to an example usage of logrotate. I tried looking through kyuubi document but could not find any information.
